src/Storages/StorageMergeTree.cpp

```diff
@@ -1489,16 +1546,16 @@ bool StorageMergeTree::scheduleDataProcessingJob(BackgroundJobsAssignee & assign
     return scheduled;
 }
 
-UInt64 StorageMergeTree::getCurrentMutationVersion(
-    const DataPartPtr & part,
-    std::unique_lock<std::mutex> & /*currently_processing_in_background_mutex_lock*/) const
-{
-    auto it = current_mutations_by_version.upper_bound(part->info.getDataVersion());
-    if (it == current_mutations_by_version.begin())
-        return 0;
-    --it;
-    return it->first;
-}
+// UInt64 StorageMergeTree::getCurrentMutationVersion(
+//     const DataPartPtr & part,
+//     std::unique_lock<std::mutex> & /*currently_processing_in_background_mutex_lock*/) const
+// {
+//     auto it = current_mutations_by_version.upper_bound(part->info.getDataVersion());
+//     if (it == current_mutations_by_version.begin())
+//         return 0;
+//     --it;
+//     return it->first;
+// }
 
 size_t StorageMergeTree::clearOldMutations(bool truncate)
```

Comment:
This one should also start to respect finished partition-local mutations

Reply:
fixed